containers/PerpetualState.ts

```diff
@@ -0,0 +1,59 @@
+import { createContainer } from "unstated-next";
+import { useState, useEffect } from "react";
+import { BigNumber, Bytes, ethers } from "ethers";
+
+import Connection from "./Connection";
+import SelectedContract from "./SelectedContract";
+import EmpContract from "./EmpContract";
+import EmpState from "./EmpState";
+import { getAbi } from "../utils/getAbi";
+
+type Provider = ethers.providers.Provider | ethers.Signer;
+
+export async function getPerpetualState(address: string, provider: Provider) {
+  const abi = getAbi("perpetual", "2");
+  const contract = new ethers.Contract(address, abi, provider);
+  return {
+    fundingRate: await contract.fundingRate(),
+    priceIdentifier: await contract.priceIdentifier(),
+  };
+}
+
+type dataType = {
+  fundingRate: {
+    rate: BigNumber;
+  };
+  priceIdentifier: string;
+};
+const useContractState = () => {
+  const { signer } = Connection.useContainer();
+  const abi = getAbi("perpetual", "2");
```

Comment:
What's 2?

Reply:
its a version number. I've set up a getAbi function which has some short names for version matching as well as contract name matching. you can see [here](https://github.com/UMAprotocol/emp-tools/blob/047163a3d2bd1f8dc674a4a8eb790c4dccd3cb28/utils/getAbi.ts#L29)